guides/TrainingModelsHPC.md

```diff
@@ -92,7 +92,25 @@
 >
 > If we launch a job and then modify the config file _before_ the job has been able to read it, we may be using an undesired version of the config in our job! To avoid this, it is best to wait until you can verify in MLflow that the job has the expected config parameters (and then edit the file to launch a new job if needed).
 
-6. **Optional argument - Optuna**
+6. **Checkpoint**
+
+   The `checkpoint_path` argument can be useful. There are two primary options related to checkpoints:
+
+   - Resume training
+
+     - This option is useful for interrupted training sessions or extending training duration.
+     - If training is disrupted and stops mid-way, you can resume it by adding `--checkpoint_path $CKPT_PATH \` to your bash script.
+     - The training will pick up from the last saved epoch and continue until the specified n_epoch.
+     - Similarly, if training completes but you want to extend it based on metric evaluations, you can increase the n_epoch value (e.g., from `n` to `n + y`). Again, use `--checkpoint_path $CKPT_PATH \` in your bash script, and training will resume from epoch `n` to `n + y`.
+     - Ensure the `save_weights_only` parameter under `checkpoint_saving` in the config file is set to `False` to resume training, as this option requires loading both weights and the training state.
+
+   - Fine-tunning
+     - This option is useful for fine-tuning a pre-trained model on a different dataset.
+     - It loads the weights from a checkpoint, allowing you to leverage pre-trained weights from another dataset.
+     - Add `--checkpoint_path $CKPT_PATH \` to your bash script to use this option.
+     - Set the `save_weights_only` parameter under `checkpoint_saving` in the config file to `True`, as only the weights are needed for fine-tuning.
```

Comment:
Some suggestions, feel free to apply as you see fit!

- I would rename the title to "Restarting training from a checkpoint", so that it is not confused with saving checkpoints during training.
- I think the main difference between the two options is that one makes use of a "full" checkpoint, and the other one uses a "weights-only" checkpoint. But you can use both to continue training right? If this is correct, I think we could simplify this part to just say this.
- I suggest moving this whole section to the main README, rather than the HPC bit. What do you think?
- Related to the bug, I think it is important to say here that even if we start training from a checkpoint, the parameters in the yaml config file apply.

Reply:
> you can use both to continue training right?
No quite, as the weights-only one is more for fine-tuning or transfer learning and is not necessarily continue, we can change the whole hyperparameter.

> I suggest moving this whole section to the main README, rather than the HPC bit
We can have them in both? We can add them later in README when we update README #192

Reply:
ok in both sounds good - I made a note in the PR!